app/components/article-content.js

```diff
@@ -388,6 +391,57 @@ export default Component.extend(
 			}
 		},
 
+		/**
+		 * Handles opening sections when click event occurs on CiteNote
+		 *
+		 * @returns {void}
+		 */
+		handleCiteNotes(event) {
+			const {target} = event;
+
+			if (target.nodeName === 'A') {
+				if (target.hash.search('#cite_ref-') === 0) {
+					const citeNoteList = this.element.querySelectorAll(target.hash);
+					let currentSection = null;
+
+					if (citeNoteList.length === 1) {
+						currentSection = citeNoteList[0].closest('section[id*="section"]');
+						if (currentSection) {
+							const currentHeader = currentSection.previousElementSibling;
+							if (currentHeader) {
+								currentHeader.classList.add('open-section');
+							}
+						}
+					}
+				}
+			}
+		},
+
+		/**
+		 * Handles opening sections when click event occurs on CiteRef
+		 *
+		 * @returns {void}
+		 */
+		handleCiteRefs(event) {
+			const {target} = event;
+
+			if (target.nodeName === 'A') {
+				if (target.hash.search('#cite_note-') === 0) {
+					const referenceListItem = this.element.querySelectorAll(target.hash);
+
+					if (referenceListItem.length === 1) {
+						const referencesSection = referenceListItem[0].parentElement.closest('section[id*="section"]');
+						let referencesHeader = null;
+
+						if (referencesSection) {
```

Comment:
this part of code looks very similar to the code from the other function
maybe you could extract openining a section to another method? :)